skimage/feature/register_translation.py

```diff
@@ -105,8 +105,36 @@ def _compute_error(cross_correlation_max, src_amp, target_amp):
     return np.sqrt(np.abs(error))
 
 
+def _triangle(N):
+    """Return N samples from the triangle function."""
+    x = np.linspace(0, 1, N, endpoint=False) + 0.5 / N
+    return 1 - abs(x - 0.5)
+
+
+def _area_overlap(A):
+    """
+    Return overlapping area of A with itself.
+
+    Create overlap arrays for higher dimensions using matrix multiplication.
+
+    >>> _area_overlap(np.empty(4))
+    array([0.625, 0.875, 0.875, 0.625])
+
+    >>> _area_overlap(np.empty((3, 5)))
+    array([[0.4       , 0.53333333, 0.66666667, 0.53333333, 0.4       ],
+           [0.6       , 0.8       , 1.        , 0.8       , 0.6       ],
+           [0.4       , 0.53333333, 0.66666667, 0.53333333, 0.4       ]])
+    """
+    for dim, shape in enumerate(A.shape):
+        if dim == 0:
+            w = _triangle(shape)
+        else:
+            w = w[..., np.newaxis] @ _triangle(shape)[np.newaxis, ...]
+    return w
+
+
 def register_translation(src_image, target_image, upsample_factor=1,
-                         space="real", return_error=True):
+                         space="real", return_error=True, reg_weight=1e-12):
```

Comment:
I may suggest to set reg_weight to zero by default though...

Reply:
On second thought, I disagree because I consider this pull request to be fixing bug not adding a feature (#4500). Please call a vote of the maintainers if necessary.

Reply:
Actually, I didn't read the issue as a bug, more as a potential improvement :)
How to you justify the default value? Does this prevent the "bug" at 100% with this default value? I guess not, so leaving to the choice of the user with a default on disable sounds a good compromise to me.

Reply:
Is your objection because you don't consider this a bug fix or because you need proof that the default value is 100% effective?

In the former case, we disagree (which is fine); edits from maintainers is allowed. In the later case, I will need conduct a study of whether `1e-12` is the most effective value or if the regularizer should be changed.